File: frontend/src/types/courier.ts
/**
 * Courier type definitions
 */

export interface Courier {
  id: string;
  name: string;
  phone: string;
  created_at: string;
  updated_at: string;
}

export interface CourierListItem {
  id: string;
  name: string;
  phone: string;
  created_at: string;
}

export interface CourierCreate {
  name: string;
  phone: string;
}

export interface CourierUpdate {
  name: string;
  phone: string;
}

export interface CourierListResponse {
  items: CourierListItem[];
  pagination: {
    page: number;
    per_page: number;
    total_items: number;
    total_pages: number;
  };
}

export interface CourierListParams {
  page?: number;
  per_page?: number;
  search?: string;
  sort_by?: 'name' | 'phone' | 'created_at';
  sort_order?: 'asc' | 'desc';
}

export interface BulkDeleteRequest {
  ids: string[];
}

export interface BulkDeleteResponse {
  deleted_count: number;
}

File: backend/app/schemas/courier.py
"""
Pydantic schemas for Courier model.
Request/response validation and serialization.
"""
from uuid import UUID
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class CourierBase(BaseModel):
    """Base courier schema with common fields."""
    name: str = Field(..., min_length=1, max_length=100)
    phone: str = Field(..., min_length=1, max_length=20)


class CourierCreate(CourierBase):
    """Schema for creating a new courier."""
    pass


class CourierUpdate(CourierBase):
    """Schema for updating an existing courier."""
    pass


class CourierResponse(CourierBase):
    """Schema for courier response (includes metadata)."""
    id: UUID
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class CourierListItem(BaseModel):
    """Schema for courier list item (lightweight version)."""
    id: UUID
    name: str
    phone: str
    created_at: datetime
    
    class Config:
        from_attributes = True


class PaginationMetadata(BaseModel):
    """Schema for pagination metadata."""
    page: int
    per_page: int
    total_items: int
    total_pages: int


class CourierListResponse(BaseModel):
    """Schema for paginated courier list response."""
    items: list[CourierListItem]
    pagination: PaginationMetadata


class BulkDeleteRequest(BaseModel):
    """Schema for bulk delete request."""
    ids: list[UUID] = Field(..., min_length=1)
